test/Swashbuckle.AspNetCore.SwaggerGen.Test/XmlComments/XmlCommentsDocumentFilterTests.cs

```diff
@@ -23,32 +25,72 @@ public void Apply_SetsTagDescription_FromControllerSummaryTags()
                         ActionDescriptor = new ControllerActionDescriptor
                         {
                             ControllerTypeInfo = typeof(FakeControllerWithXmlComments).GetTypeInfo(),
-                            ControllerName = nameof(FakeControllerWithXmlComments)
+                            ControllerName = nameof(FakeControllerWithXmlComments),
+                            RouteValues = new Dictionary<string, string> { { "controller", nameof(FakeControllerWithXmlComments) } }
                         }
                     },
                     new ApiDescription
                     {
                         ActionDescriptor = new ControllerActionDescriptor
                         {
                             ControllerTypeInfo = typeof(FakeControllerWithXmlComments).GetTypeInfo(),
-                            ControllerName = nameof(FakeControllerWithXmlComments)
+                            ControllerName = nameof(FakeControllerWithXmlComments),
+                            RouteValues = new Dictionary<string, string> { { "controller", nameof(FakeControllerWithXmlComments) } }
                         }
                     }
                 },
                 null,
                 null);
 
-            Subject().Apply(document, filterContext);
+            Subject(options).Apply(document, filterContext);
 
             var tag = Assert.Single(document.Tags);
             Assert.Equal("Summary for FakeControllerWithXmlComments", tag.Description);
         }
 
-        private static XmlCommentsDocumentFilter Subject()
+        [Fact]
+        public void Uses_Proper_Tag_Name()
+        {
+            var options = new SwaggerGeneratorOptions();
```

Comment:
Could we add a test with a custom tag selector as well?

Reply:
Done